### ISSUE TITLE
DNS proof failing

### ISSUE THREAD

**Issue by adamreisnz** [keybase/keybase-issues#2700]
I'm trying to proof my domain, `adam.reis.nz` but it keeps failing after more than 24 hours.

DNS check: https://dnschecker.org/#TXT/reis.nz

![image](https://cloud.githubusercontent.com/assets/490562/20878422/b78cc718-bb33-11e6-8cf6-459e9c947566.png)

The proof is there, as directed by the site. I have tried to replace it already but still doesn't work. Please advise.

**Comment by Darael** [comment]
...you are checking TXT records for reis.nz but you say you are trying to prove adam.reis.nz.  If the latter is truly the case then keybase will be doing DNS lookups for "adam.reis.nz in TXT" and "_keybase.adam.reis.nz in TXT", and neither of these will return the record which the propagation-check says is "reis.nz in TXT".

**Comment by adamreisnz** [comment]
@Darael in that case the instructions are wrong, as they state:

![image](https://cloud.githubusercontent.com/assets/490562/20898963/4b8d73d2-bb8d-11e6-8767-9bc56ce98d86.png)

The `@` symbol refers to the root domain. Keybase should realize I'm trying to proof a *sub*domain, and ask me to create the TXT record for the `adam` name.

I will try again.

**Comment by adamreisnz** [comment]
Yep, now it says I succeeded. So looks like you need to add in a check to display the correct instructions if someone enters a subdomain.

**Comment by Darael** [comment]
That's a bug in the graphical interface.  I've just checked, and the command-line version gives correct instructions:

```
$ keybase prove dns
Your DNS domain: darael.dracon.is
Please save the following as a DNS TXT entry for darael.dracon.is OR _keybase.darael.dracon.is:

keybase-site-verification=(etc)

Check Dns now? [Y/n]
```

**Comment by Darael** [comment]
The thing is it doesn't know what your DNS zone is.  For all the keybase client knows, adam.reis.nz is delegated to a separate nameserver from the rest of reis.nz, where the root zone as referred to by @ *would* be adam.reis.nz.  It should just give the full domain, as the command-line version does, and let the person doing the proof, who is expected to know about managing DNS, work out exactly which name to use for the TXT record.
